workflow-dev/tpl/minio.sh

```diff
@@ -1,12 +1,12 @@
 #!/bin/bash
-while IFS='' read -r line || [[ -n "$line" ]]; do
-  if [[ $line == "storage"* ]]; then
-    IFS='=' read -a storage <<< "$line"
-    if [ ${storage[1]} == "\"minio\"" ]; then
-      helmc tpl -d $HELM_GENERATE_DIR/tpl/generate_params.toml -o $HELM_GENERATE_DIR/manifests/deis-minio-rc.yaml $HELM_GENERATE_DIR/tpl/deis-minio-rc.yaml
-      cp $HELM_GENERATE_DIR/tpl/deis-minio-service* $HELM_GENERATE_DIR/manifests/
-    else
-      rm -rf $HELM_GENERATE_DIR/manifests/deis-minio-*
-    fi
+store=`grep "storage = " $HELM_GENERATE_DIR/tpl/generate_params.toml |cut -d "=" -f2 | tr -dc '[:alnum:]\n\r'`
+if [ -n "$STORAGE_TYPE" ]; then
+  if [ "$STORAGE_TYPE" != "minio" ]; then
+    rm -rf $HELM_GENERATE_DIR/manifests/deis-minio-*
   fi
-done < "$HELM_GENERATE_DIR/tpl/generate_params.toml"
+elif [ "$store" != "minio" ]; then
+  rm -rf $HELM_GENERATE_DIR/manifests/deis-minio-*
```

Comment:
Why do we need to actively destroy this file?

Assuming this logic is needed, is it possible to collapse the two conditions into one statement?  in pseudocode: `if $STORAGE_TYPE != "minio" OR $store != "minio", remove the correct file`